codegen/facelift/templates/QMLFrontend.template.h

```diff
@@ -217,15 +217,15 @@ class {{class}}QMLFrontend : public facelift::QMLFrontendBase
     }
     {% else %}
 
-    Q_INVOKABLE {{operation.cppType}} {{operation}}(
+    Q_INVOKABLE {{operation.type.qmlCompatibleType}} {{operation}}(
         {%- set comma = joiner(", ") -%}
         {%- for parameter in operation.parameters -%}
         {{ comma() }}{{parameter.type.qmlCompatibleType}} {{parameter.name}}
         {%- endfor -%}
     )
     {
         Q_ASSERT(m_provider);
-        return m_provider->{{operation}}(
+        return {% if operation.hasReturnValue %}facelift::toQMLCompatibleType{% endif %}(m_provider->{{operation}}(
```

Comment:
Return value check could be moved before 'return' statement (a void function with return is weird)